internal/core/src/query/generated/ExecExprVisitor.h

```diff
@@ -64,9 +64,9 @@ class ExecExprVisitor : public ExprVisitor {
     auto
     ExecRangeVisitorImpl(FieldId field_id, IndexFunc func, ElementFunc element_func) -> BitsetType;
 
-    template <typename T, typename ElementFunc>
+    template <typename T, typename IndexFunc, typename ElementFunc>
     auto
-    ExecDataRangeVisitorImpl(FieldId field_id, ElementFunc element_func) -> BitsetType;
+    ExecDataRangeVisitorImpl(FieldId field_id, IndexFunc index_func, ElementFunc element_func) -> BitsetType;
```

Comment:
Shoule we rename this method since we'll change it's function?

Reply:
By the way, what's the difference between this method and `ExecRangeVisitorImpl`.

Reply:
> By the way, what's the difference between this method and `ExecRangeVisitorImpl`.

ExecDataRangeVisitorImpl will use raw data for cal firstly, which is useful for growing segment.

Reply:
OK, I think I got what you mean. `ExecDataRangeVisitorImpl` will just be used on arithmetic eval range filterings. These filterings actually are better to be executed by bruteforce-scan. If raw-data was loaded, for sealed segments, the execution won't be applied on index, otherwise won't be applied on raw-data.